R/loc_median_fit.R

```diff
@@ -27,39 +29,64 @@
 #'
 #'   plot(x, y)
 #'   fit <- loc_median_fit(x, y, fraction = 0.1)
+#'   fit2 <- loc_median_fit(x, y, fraction = 0.1, sample_fraction = 0.75)
 #'   points(x, fit, col = "red")
+#'   points(x, fit, col = "blue")
 #'
 #'
 #' @export
 loc_median_fit <- function(x, y, fraction = 0.1, npoints = max(1, round(length(x) * fraction)),
-                           weighted = TRUE, ignore_zeros = FALSE){
+                           weighted = TRUE, ignore_zeros = FALSE, sample_fraction = 1, seed=42){
+  if (length(x) != length(y)) {
+    stop("x and y must be of the same length.")
+  }
+  if (length(x) == 0) {
+    return(numeric(0L))
+  }
+  if (!is.numeric(sample_fraction) || sample_fraction <= 0 || sample_fraction > 1) {
+    stop("sample_fraction must be a numeric value between 0 and 1.")
+  }
+  if (is.null(npoints)) {
+    npoints <- max(1, round(length(x) * fraction))
+  }
+  if (npoints <= 0 || npoints > length(x)) {
+    stop("npoints must be a positive integer less than or equal to the length of x.")
+  }
+
+  if(sample_fraction != 1){
+    set.seed(seed)
+    subset_size <- round(sample_fraction * length(x))
+    sample_indices <- sample(1:length(x), size = subset_size)
```

Comment:
Please use `seq_along(x)` instead of `1:length(x)`. Even better would be to call `sample.int(length(x), size = subset_size)`.